### ISSUE TITLE
Receiver/Tag timelines showing the wrong batches in some cases

### ISSUE THREAD

**Issue by zcrysler** [MotusDev/Motus-TO-DO#387]
Three receivers registered to a European project (113), are showing American detections back in 2015/2016. These European receivers were not swapped between projects so it looks like another case of duplicate serial numbers. It likely won't cause issues since it seems like the American deployments are over, but they should probably still be dealt with the same as the other duplicates in case it comes up again in the future. The American receivers are all registered to project 0:

•	SG-5016BBBK17DA 
•	SG-5016BBBK19C0
•	SG-5016BBBK200B

**Comment by Hussell** [comment]
I'm not seeing any detections prior to 2018 in the database, on the receiver timelines, or in the sgdata receiver plots. Please give a link to the data from 2015/2016.

**Comment by zcrysler** [comment]
When you look at the timelines accessed from the receivers page it doesn't show them (https://motus.org/data/receiver/timeline?id=2450&year=2015), but when you access from explore data > projects > receivers > timeline it shows them: https://motus.org/data/timeline?receiverDeploymentId=4715

**Comment by Hussell** [comment]
Not a duplicate receiver issue. Seems to be a bug in the timeline page, possibly a failure to filter out duplicate batches.

**Comment by cjardine-bsc** [comment]
What view does the timeline page use?

**Comment by Hussell** [comment]
The code for these pages is quite old, from before the first CANARIE grant. It looks like it's pulling directly from sg_taghits_runs. I'm looking at rewriting the queries to use the views now.

**Comment by cjardine-bsc** [comment]
*confetti*

**Comment by Hussell** [comment]
Ok, while the queries *did* need to be rewritten, that turns out not to have been the problem. Neither are the views.

Try this:
select *
from sg_taghits_runs a
join sensor_deployments b on b.id = a.sensor_deploy_id
where a.sensor_deploy_id = 4715
and a.max_ts < datediff(s, '1970', b.dt_start)

On the main database I get 1997 rows, and on sandbox I get 0. How is sg_taghits_runs populated?

**Comment by denislepage** [comment]
This is populated at the time the data is imported from sgdata, normally. We should check how many are missing, and have some sort of fail-safe.

**Comment by Hussell** [comment]
To clarify, the issue isn't data missing from sandbox, it's bad data in the live database. For example there are two batches in sg_taghits_runs (100072 and 100079) which supposedly have runs for sensor deployment 4715, but the runs start and end before the deployment. Checking into those batches, sg_taghits_runs claims they contain runs for 161 different deployments each, which is improbable. The table in sandbox looks much saner (only one deployment for each batch), so I'm guessing the problem is due to a change since the last time the DBs were synced. It looks like it's just those two batches. Batch 130220 involves 5 deployments, but that's the only other one with more than 3 deployments.

I think the relevant code might be in prTaghitsSummariesAll and prTaghitsSummariesBatchID, called from org.bsc.motus.process.SGDataImport. Still not sure what the bug is.

**Comment by cjardine-bsc** [comment]
This is related to the big re-run of prTaghitsDeployIDSensorDeployID that we did to address the issue in that proc where a random deviceID was being returned by a subquery (a qualifier was missing). I discovered this when I noticed an issue with deviceID 98, which seems to be the deviceID the query tended to return.  (Denis see my email from Aug 1st)

We did the update to fix all the deploy_id tables, but we still need to update the summaries to fix the bad data that went into those summaries when the SensorDeployID proc was broken. 

The prTaghitsSummariesBatchID proc is working correctly (for example if you feed it batch 100072 now it would return only 91 records for sg_taghits_runs with sensor deploy id = 2176 instead of 4715.

The total number of records in sg_taghits_runs for batch 100072 right now is 14651. Which is 161 times the 91 records that should be there. Likely meaning that 161 deployments were created or edited while the proc was bad an inserted into this table in error. 

The issue isn't in sandbox because we don't sync the sg tables between the two DBs. And no one was creating deployments and triggering prTaghitsDeployIDSensorDeployID on sandbox. 

So what we need to do is re-run prTaghitsSummaries. I *think* we only need to do that for batches for deviceID 98, but I can't be 100% sure because I don't fully know what logic SQL used to select the top 1 record into a variable when given it a multi-record list. 

I've re-run it now for the batches for deviceID 98, but I'd like to advocate that we do a complete re-run. It's going to take a while to run, and it will mean the summaries will be blank on the motus site for a little while as we wipe and replace them all, but I think it's worth it. We've fixed quite a few DB bugs recently and a complete wipe and replace of the summaries is probably prudent. Denis, I await your blessing.
